core/src/com/unciv/ui/screens/worldscreen/minimap/MinimapHolder.kt

```diff
@@ -55,13 +67,72 @@ class MinimapHolder(val mapHolder: WorldMapHolder) : Table() {
         this.clear()
         minimap = Minimap(mapHolder, minimapSize, civInfo)
         val wrappedMinimap = getWrappedMinimap()
-        add(getToggleIcons(wrappedMinimap.height)).align(Align.bottom)
-            .height(wrappedMinimap.height).padRight(5f) // Spread equally over the side
-        add(wrappedMinimap)
+        add(getToggleIcons(wrappedMinimap.height)).bottom().padRight(5f)
+        
+        val stack = Stack()
+        stack.add(wrappedMinimap)
+        stack.addInTable(getCornerHandleIcon()).size(20f).pad(8f).top().left()
+        stack.addInTable(getMaximizeToggleButton(civInfo)).size(40f).bottom().right() // more click area
+        add(stack).bottom()
+        
         pack()
         if (stage != null) x = stage.width - width
+        
+        addListener(ResizeDragListener(civInfo))
+    }
+
+    private fun rebuildAndUpdateMap(civInfo: Civilization?) {
+        rebuild(civInfo) // re-create views
+        civInfo?.let { minimap.update(it) } // update map
+        minimap.mapHolder.onViewportChanged() // update scroll position
     }
 
+    private fun getMaximizeToggleButton(civInfo: Civilization?): Actor {
+
+        // when maximized, collapse map when a location was clicked
+        if (maximized) {
+            minimap.onClick { maximized = false }
+        }
+
+        val toggle = fun() {
+            maximized = !maximized
+            minimapSize = if (maximized && stage!=null) {
+                minimap.getClosestMinimapSize(Vector2(stage.width, stage.height)) - 2
+            } else {
+                worldScreen.game.settings.minimapSize
+            }
+            rebuildAndUpdateMap(civInfo)
+        }
+
+        val name = if(maximized) "Reduce" else "Increase"
+        val image = ImageGetter.getImage("OtherIcons/$name")
+        // table provides larger click area
+        return Table().apply {
+            add(image).expand().size(20f).pad(8f).bottom().right()
+            touchable = Touchable.enabled
+            onActivation(toggle)
+            // map is really small: use whole minimap area as click area to maximize map
+            if (!shouldShowMapButtons()) {
+                isVisible = false
+                minimap.touchable = Touchable.disabled
+                this@MinimapHolder.apply { 
+                    touchable = Touchable.enabled
+                    onClick(toggle)
```

Comment:
So... If we call getMaximizeToggleButton twice, then two toggle actions would have been added to the minimap holder
Does this mean they'll both activate and cancel out?

Reply:
No because in that case the button is invisible. But I'll restructure that part a bit to make it less confusing :+1: